client/src/pages/SignUp.js

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { signup, login } from '../store/auth';
+import { Redirect } from 'react-router-dom';
+import { Container } from '@material-ui/core'
+
+import '../style/signup.css'
+
+function SignUp() {
+    const [username, setUsername] = useState('')
+    const [email, setEmail] = useState('')
+    const [password, setPassword] = useState('')
+    const [password2, setPassword2] = useState('')
+    const currentUserId = useSelector(state => state.auth.id);
+    const dispatch = useDispatch();
+    
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        dispatch(signup(username, email, password, password2))
+        dispatch(login(username, password))
```

Comment:
You might need to test this locally to check, but I don't think `dispatch(login(username, password))` is needed.

It looks like when you dispatch the `signup` function, you're return `action.user` in the `authReducer`, meaning that the `state.auth.id` should be updated and cause the `if (currentUserId) return <Redirect to="/profile" />;` line of code to run.

Let me know if it doesn't actually redirect you after removing the `login` dispatch call!